shared-bindings/ps2io/Ps2.c

```diff
@@ -118,50 +118,69 @@ STATIC mp_obj_t ps2io_ps2_obj___exit__(size_t n_args, const mp_obj_t *args) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(ps2io_ps2___exit___obj, 4, 4, ps2io_ps2_obj___exit__);
 
-//|   .. method:: get_byte()
+//|   .. method:: popleft()
 //|
 //|     Removes and returns the oldest received byte. When buffer
-//|     is empty, returns a negative number.
+//|     is empty, raises an IndexError exception.
 //|
-STATIC mp_obj_t ps2io_ps2_obj_get_byte(mp_obj_t self_in) {
+STATIC mp_obj_t ps2io_ps2_obj_popleft(mp_obj_t self_in) {
     ps2io_ps2_obj_t *self = MP_OBJ_TO_PTR(self_in);
     raise_error_if_deinited(common_hal_ps2io_ps2_deinited(self));
 
-    return MP_OBJ_NEW_SMALL_INT(common_hal_ps2io_ps2_get_byte(self));
+    int b = common_hal_ps2io_ps2_popleft(self);
+    if (b < 0) {
+        mp_raise_IndexError(translate("pop from an empty Ps2 buffer"));
+    }
+    return MP_OBJ_NEW_SMALL_INT(b);
 }
-MP_DEFINE_CONST_FUN_OBJ_1(ps2io_ps2_get_byte_obj, ps2io_ps2_obj_get_byte);
-
-//|   .. method:: send_byte()
-//|
-//|     Sends a byte to PS/2. Returns the response byte (positive)
-//|     or a communication error code (negative). The response is
-//|     typically an ACK (0xFA). Some commands return additional data
-//|     bytes that should be received via get_byte().
-//|
-STATIC mp_obj_t ps2io_ps2_obj_send_byte(mp_obj_t self_in, mp_obj_t ob) {
+MP_DEFINE_CONST_FUN_OBJ_1(ps2io_ps2_popleft_obj, ps2io_ps2_obj_popleft);
+
+//|   .. method:: sendcmd()
+//|
+//|     Sends a command byte to PS/2. Returns the response byte.
+//|     The response is typically an ACK (0xFA).
+//|     Some commands return additional data  bytes that should be
+//|     received via popleft().
+//|     Raises an OSError in case of error. Possible error values:
+//|        0x01: clock pin didn't go to LO in time
+//|        0x02: clock pin didn't go to HI in time
+//|        0x04: data pin didn't ack
+//|        0x08: clock pin didn't ack
+//|        0x10: device didn't respond to Request to Send
+//|        0x20: device didn't send response byte in time
+STATIC mp_obj_t ps2io_ps2_obj_sendcmd(mp_obj_t self_in, mp_obj_t ob) {
     ps2io_ps2_obj_t *self = MP_OBJ_TO_PTR(self_in);
     raise_error_if_deinited(common_hal_ps2io_ps2_deinited(self));
-    mp_int_t b = mp_obj_get_int(ob) & 0xff;
-    return MP_OBJ_NEW_SMALL_INT(common_hal_ps2io_ps2_send_byte(self, b));
+    mp_int_t cmd = mp_obj_get_int(ob) & 0xff;
+    int resp = common_hal_ps2io_ps2_sendcmd(self, cmd);
+    if (resp < 0) {
+        mp_raise_OSError(-resp);
```

Comment:
Please don't use OSError because its values already have specific meaning. I only gave it as an example of an exception with state.

Is it important that the reason be programmatically available? If not, you could just throw RuntimeError with a number of different messages. If it is, you can return RuntimeError here and state that the one should call get_errors for details.